src/main/java/edu/kit/datamanager/repo/util/validators/impl/HandleNetValidator.java

```diff
@@ -0,0 +1,156 @@
+import edu.kit.datamanager.repo.util.validators.IValidator;
+import org.datacite.schema.kernel_4.RelatedIdentifierType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class validates Handles with help of handle.net.
+ */
+public class HandleNetValidator implements IValidator {
+
+    Logger log = LoggerFactory.getLogger(HandleNetValidator.class);
+
+    /**
+     * This method returns the type of the validator implementation.
+     *
+     * @return element of the enum defined in the Datacite schema.
+     */
+    @Override
+    public RelatedIdentifierType supportedType() {
+        return RelatedIdentifierType.HANDLE;
+    }
+
+    /**
+     * This method must be implemented by any implementation.
+     * It validates an input and either returns true or throws an exception.
+     *
+     * @param input to validate
+     * @param type  of the input
+     * @return true if input is valid for the special type of implementation.
+     */
+    @Override
+    public boolean isValid(String input, RelatedIdentifierType type){
+        if (type != supportedType()) {
+            LOG.warn("Illegal type of validator");
+            throw new UnsupportedMediaTypeException("Illegal type of Validator.");
+        }
+
+        String regex = "^(hdl://|http://|https://|doi:)(.+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(input);
+        if (matcher.find()) {
+            if (matcher.group(1).equals("https://") || matcher.group(1).equals("http://"))
+                return isValidHTTPURL(matcher.group(0));
+            else return isValidHandle(matcher.group(2));
+        } else return isValidHandle(input);
+    }
+
+    /**
+     * This method checks if the given prefix and suffix is downloadable.
+     * If there is some invalid input, it will validate the prefix with help of the RESTful API by handle.net.
+     *
+     * @param serverAddress from the server which should be uses for the validation.
+     * @param prefix        the handle prefix
+     * @param suffix        the handle suffix
+     * @return true if the record is downloadable
+     */
+    private boolean isDownloadable(String serverAddress, String prefix, String suffix) {
+        IValidator urlValidator = new URLValidator();
+        AtomicBoolean fullValid = new AtomicBoolean(false);
+        log.debug("Server address: {}", serverAddress);
+        log.debug("Prefix: {}", prefix);
+        log.debug("Suffix: {}", suffix);
+        ignoringExc(() -> {
+            try{
+                fullValid.set(urlValidator.isValid(serverAddress + "/" + prefix + "/" + suffix));
+            } catch (Exception ignored){
+            }
+        });
+        if (fullValid.get()) {
+            LOG.info("The handle is valid!");
```

Comment:
Info level seems not reasonable. (is this information of interest? I don't think so.) I would use debug instead as the logs before.

Reply:
I disagree!
IMO Level 'Info' should log only information about the internal state of the service.
e.g.: debug level set, services connected, available validators, ... (should be logged once at startup) 
If there are thousands of digital objects ingested to the repo it's annoying to see that it always went right. 
If something goes wrong use 'WARN' or 'ERROR'. If everything is alright this should be only visible in debug mode.

Reply:
If you want to set different levels you may also use 'TRACE'. ;-)